### PR TITLE
[lgwebos] Mask access key

### LINKED ISSUE #7758: [lgwebos] Access key
The access key is currently not considered by the binding as a personal data to protect:

- the configuration setting is not masked in Paper UI (configuration setting not defined as a password)
- the messages containing this access key are logged

Not masking this configuration setting in Paper UI has an advantage: you can easily retrieve it and then use it in an openHAB config file to define your thing.

I imagine two solutions.

Solution 1

- We define this configuration setting as a password so it will then be masked in Paper UI
- When a message containing this access key is logged, we replace the value of the field "client-key" by "***" in the JSON message before logging it
- As a workaround to retrieve this access key, we log it (only if it was not yet set for the thing).

Solution 2
- We let this configuration setting unmasked in Paper UI
- When a message containing this access key is logged, we replace the value of the field "client-key" by "***" in the JSON message before logging it

@sprehn : WDYT ?

See #3990

### PR DESCRIPTION
**Pull request by lolodomo**
Fix #7758

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
This first part allows not logging the access key when contained in the exchanged messages.

Just remains to decide if we set the configuration setting as "password" or not.

File: bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSHandler.java [hunk 1]
```diff
@@ -256,7 +256,7 @@ public String getKey() {
     @Override
     public void storeKey(@Nullable String key) {
         if (!getKey().equals(key)) {
-            logger.debug("store new key");
+            logger.info("Store {} as access Key in the thing configuration", key);
```
Comment (J-N-K) [inline bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSHandler.java:259]: If the key is sensitive information, please do not log it at all. Maybe log the number of characters.

Comment (lolodomo) [reply bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSHandler.java:259]: We need to give a chance to the user to get the value of the access key. The user will need it to set up his thing in a config file.

Please note that this log will appear only when the access key is not yet defined in the thing configuration, that means when first setting up the thing. As soon as the access key is set in the thing configuration, this code will no more be reached and so the access key will no more be logged.

This is done like that in few other bindings, when the key is a required configuration setting which is provided by the device at paring time (hue or freebox bindings for example).

One alternative would be to not log the access key as this place and make the access key configuration setting always visible in Paper UI. But in this case, the access key is ALWAYS accessible in Paper UI. I think it is less secure than the solution I propose.

Comment (J-N-K) [reply bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSHandler.java:259]: Agreed. We can leave it like that. I would prefer a different strategy, e.g. a console command that can print the current access key from memory. That way we could prevent logging at all. But that is out of scope of this PR.

Comment (lolodomo) [reply bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSHandler.java:259]: I agree with you, the console command is a good idea. I will add this feature.

**Pull request merged**